src/foam/dao/java/FindReplayDAO.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package foam.dao.java;
+
+import foam.dao.DAO;
+import foam.dao.ProxyDAO;
+import foam.core.X;
+
+/**
+ * This DAO is a decorator around the innerDAO to return the DAO when replaying
+ * entries directly into the DAO.
+ */
+public class FindReplayDAO
+  extends ProxyDAO {
+
+  public final static Object FIND_REPLAY_DAO_CMD = new Object();
```

Comment:
Can you use the context instead to store a 'replayDAO' that RoutingJournal can look up? Then I don't think you'd need this FindReplayDAO at all.